src/plugins/data/public/ui/query_string_input/query_string_input.tsx

```diff
@@ -549,10 +549,13 @@ export class QueryStringInputUI extends Component<Props, State> {
     window.removeEventListener('scroll', this.handleListUpdate);
```

Comment:
I think we never clean up this listener. 

We register it as `capture`:

```
window.addEventListener('scroll', this.handleListUpdate, {
      passive: true, // for better performance as we won't call preventDefault
      capture: true, // scroll events don't bubble, they must be captured instead
    });
```

Which means we should also remove it as `capture`: 

```
window.removeEventListener('scroll', this.handleListUpdate, {capture: true});
```

Created a sandbox to double check and indeed seems like it never cleaned up: https://codesandbox.io/s/heuristic-hooks-gnu70?file=/src/index.js

Reply:
But in addition to that, you initial improvement also makes total sense, so I'd leave it also 👍

Reply:
TIL!